webpack.prod.config.ts

```diff
@@ -40,5 +41,14 @@ module.exports = merge(common, {
         analyticsConstants.SITE_NAME_FOR_ANALYTICS
       ),
     })
-  ]
+  ],
+  optimization: {
+    minimizer: [
+      new TerserPlugin({
+        cache: true,
+        parallel: false,
+        sourceMap: true,
+      }),
+    ],
+  }
```

Comment:
I took a detailed look into it and looks like it's overriding the optimization config from https://github.com/oppia/oppia/blob/develop/webpack.dev.config.ts
```
{
    runtimeChunk: 'single',
    sideEffects: true,
    usedExports: true,
    splitChunks: {
      chunks: 'all'
    },
  }
  ```
  
  (not sure!)
  
  Also, why do we need these changes which were not needed earlier?

Reply:
using Terser plugin, I added parallelization property setting it to false so I cant understand how it is overriding the above stated changes. also you stated changes from dev.config.ts, but I added configs in prod.config.ts so that we can have parallelization as false in prod mode. 

the changes I added fixes the prod mode in docker setup. the issue was in docker, prod mode was taking infinite time to start. on multiple discussion with @vojtechjelinek, he suggested to test after using this terser plugin and set the parallel property false which fixes the prod mode under docker. 
please let me know if you are asking about something different

Reply:
(I had shared wrong link earlier) I've updated the link, PTAL!

(Note: webpack.prod.configg.ts fil;es uses the common config file so I'm not sure whether the changes override the optimization section from common config file, can you please confirm?)

Reply:
the Terser plugin is a third party plugin. note that the properties in the common.config optimisation and the terser plugin are different so ideally it should not override, it should maintain properties listed in both common.config and the terser plugin properties. but I am finding some documentation supporting my statement

Reply:
@Shivkant-Chauhan CAn you pelase update this thread with out offline discussion?

Reply:
since the webpack merge can override the same key "optimisation", added the common.config's optimisation to the prod.config also by which all the optimisations are concatenated.
thanks @DubeySandeep for pointing this out and helping in it